src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -745,18 +797,42 @@ await ExecuteWithFailOverPolicyAsync(
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+
+                List<KeyValueWatcher> selectedKeyValueWatchers = new List<KeyValueWatcher>();
+
+                if (_options.RefreshOptions.RegisterAllEnabled)
+                {
+                    foreach (KeyValueIdentifier keyValueIdentifier in watchedSelectedKeyValueCollections.Keys)
+                    {
+                        selectedKeyValueWatchers.AppendUnique(new KeyValueWatcher()
+                        {
+                            Key = keyValueIdentifier.Key,
+                            Label = keyValueIdentifier.Label,
+                            RefreshInterval = _options.RefreshOptions.RefreshInterval
+                        });
+                    }
+                }
+
+                // Update the next refresh time for all refresh registered settings and feature flags
+                foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers.Concat(_options.FeatureFlagWatchers).Concat(selectedKeyValueWatchers))
+                {
+                    UpdateNextRefreshTime(changeWatcher);
+                }
+
                 _watchedSettings = watchedSettings;
+                _watchedSelectedKeyValueCollections = watchedSelectedKeyValueCollections;
+                _watchedFeatureFlagCollections = watchedFeatureFlagCollections;
                 _mappedData = mappedData;
+                _selectedKeyValueWatchers = selectedKeyValueWatchers;
             }
         }
 
-        private async Task<Dictionary<string, ConfigurationSetting>> LoadSelectedKeyValues(ConfigurationClient client, CancellationToken cancellationToken)
+        private async Task<Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>> LoadSelected(Dictionary<string, ConfigurationSetting> existingData, IEnumerable<KeyValueSelector> selectors, ConfigurationClient client, CancellationToken cancellationToken)
         {
-            var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>> watchedCollections = new Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>();
 
-            // Use default query if there are no key-values specified for use other than the feature flags
-            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => selector.KeyFilter == null ||
-                !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
+            // Use default query if there are no key-values specified for use other than feature flags
+            bool useDefaultQuery = !_options.KeyValueSelectors.Any();
```

Comment:
Variable not needed if the condition is so simple.

Reply:
Got it, though this helped me realize that I think this unnecessarily does the default query twice since I added one call to `LoadSelected` for kvs and one for ffs. I think I could add a default selector for kvs before this method so we can skip this check entirely, or if we want to keep this check I'm not sure how to restructure it

Reply:
Moved default query logic to options class